internal/storage/v2/grpc/client.go

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) 2025 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package grpc
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"iter"
+
+	"go.opentelemetry.io/collector/pdata/ptrace"
+	"google.golang.org/grpc"
+
+	"github.com/jaegertracing/jaeger/proto-gen/storage/v2"
+)
+
+type Client struct {
+	readerClient storage.TraceReaderClient
+}
+
+func NewClient(tracedConn *grpc.ClientConn) *Client {
+	return &Client{
+		readerClient: storage.NewTraceReaderClient(tracedConn),
+	}
+}
+
+func (c *Client) GetTraces(ctx context.Context,
+	traceIDs ...storage.GetTraceParams,
+) iter.Seq2[[]ptrace.Traces, error] {
+	return func(yield func([]ptrace.Traces, error) bool) {
+		query := []*storage.GetTraceParams{}
+		for _, traceID := range traceIDs {
+			query = append(query, &traceID)
+		}
+		stream, err := c.readerClient.GetTraces(ctx, &storage.GetTracesRequest{
+			Query: query,
+		})
+		if err != nil {
+			yield(nil, fmt.Errorf("received error from grpc reader client: %w", err))
+			return
+		}
+		for received, err := stream.Recv(); !errors.Is(err, io.EOF); received, err = stream.Recv() {
+			if err != nil {
+				yield(nil, fmt.Errorf("received error from grpc stream: %w", err))
+				return
+			}
+			traces := []ptrace.Traces{received.ToTraces()}
```

Comment:
this looks odd. Does the returned chunk contain a single `ptrace.Traces`? If it does, it means that if the backend needs to return multiple traces then it will have to break them into separate chunks, even though efficiency-wise it might be better to group then into a single payload. It's for this reason the tracestore.Reader API returns sequence of slices, not a sequence of `ptrace.Traces`, because of the declared convention that a single instance `ptrace.Traces` must not contain different trace IDs.

Reply:
@yurishkuro Yeah it does return a single `ptrace.Traces` at the moment. This is how we've defined the `GetTraces` call at the moment. Do we want to change this? 

```proto
rpc GetTraces(GetTracesRequest) returns (stream opentelemetry.proto.trace.v1.TracesData) {}
```

Reply:
I see, this is how api_v3 is defined. which is a bit legacy at this point. I am worried that this forces an inefficiency at the protocol level. What would it take to return a stream of wrappers instead that contain `TracesData`?

Reply:
@yurishkuro https://github.com/jaegertracing/jaeger/pull/6830